File: go/security.go
package common

import("crypto/hmac";"crypto/sha256";"crypto/subtle";"encoding/base64";"strings")
func VerifyHMACSHA256(body []byte, secret, provided, prefix string) bool { actual:=strings.TrimPrefix(provided,prefix);mac:=hmac.New(sha256.New,[]byte(secret));mac.Write(body);expected:=base64.StdEncoding.EncodeToString(mac.Sum(nil));if len(expected)!=len(actual){return false};return subtle.ConstantTimeCompare([]byte(expected),[]byte(actual))==1 }

File: php/src/security.php
<?php

declare(strict_types=1);

function verify_hmac_sha256(string $body, string $secret, string $provided, string $prefix = ''): bool
{
    $actual = str_starts_with($provided, $prefix) ? substr($provided, strlen($prefix)) : $provided;
    return hash_equals(base64_encode(hash_hmac('sha256', $body, $secret, true)), $actual);
}
